### PR TITLE
Problem: no integration test for websocket rpc endpoint

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #364
Solution:
- add cronos-ws provider to run exiting test cases on websocket endpoint
- add websocket subscription test cases
- update ethermint to fix websocket bugs
- port a log redirection feature

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -164,7 +164,7 @@ replace (
 	// See https://github.com/tecbot/gorocksdb/pull/216
 	github.com/tecbot/gorocksdb => github.com/cosmos/gorocksdb v1.1.1
 
-	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-alpha1-cronos-6
+	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-alpha1-cronos-6.0.20220301081024-4ec8b052c769
```
Comment (yihuang) [inline go.mod:167]: https://github.com/crypto-org-chain/ethermint/compare/v0.10.0-alpha1-cronos-6...4ec8b052c769?expand=1

File: integration_tests/test_basic.py [hunk 2]
```diff
@@ -30,7 +30,6 @@
 def test_basic(cluster):
     w3 = cluster.w3
     assert w3.eth.chain_id == 777
-    assert w3.eth.get_balance(ADDRS["community"]) == 10000000000000000000000
```
Comment (yihuang) [inline integration_tests/test_basic.py:33]: it only works if this is the very first test case to run, but it's not guaranteed.

File: integration_tests/conftest.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def geth(tmp_path_factory):
     yield from setup_geth(path, 8545)
 
 
-@pytest.fixture(scope="session", params=["cronos", "geth"])
+@pytest.fixture(scope="session", params=["cronos", "geth", "cronos-ws"])
```
Comment (yihuang) [inline integration_tests/conftest.py:50]: all the test cases which use the `cluster` fixture will be run an extra time against Cronos WebSocket endpoint.

**Review by JayT106 — APPROVED** [review]
LGTM

**Pull request merged**